OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitYoutubePlayerFragment.java

```diff
@@ -50,88 +59,97 @@ public static CourseUnitYoutubePlayerFragment newInstance(VideoBlockModel unit)
         return fragment;
     }
 
-    @Override
-    public void onActivityCreated(Bundle savedInstanceState) {
-        super.onActivityCreated(savedInstanceState);
-        releaseYoutubePlayer();
-        if (VideoUtil.isYoutubeAPISupported(getContext())) {
-            youTubePlayerFragment = new YouTubePlayerSupportFragment();
-            getChildFragmentManager().beginTransaction().replace(R.id.player_container, youTubePlayerFragment, "player").commit();
-        }
-        attempts = 0;
-    }
-
     @Override
     public void onResume() {
         super.onResume();
         if (unit != null) {
             setVideoModel();
-            /*
-             * This method is not called property when the user leaves quickly the view on the view pager
-             * so the youtube player can not be released( only one youtube player instance is allowed by the library)
-             * so in order to avoid to create multiple youtube player instances, the youtube player only will be initialize
-             * after minor delay.
-             */
-            initializeHandler.post(this::initializeYoutubePlayer);
+            initializeYoutubePlayer();
```

Comment:
I have some reservations about this approach - Initializing the player on `onResume`
- It takes too much time and resources
- It creates a glitch effect because we remove/add the view every time we initialize the player

What we can do:
1. Set the viewModel & Initialize the player in the `onViewCreated()` method.
2. Add the [observer](https://github.com/PierfrancescoSoffritti/android-youtube-player#lifecycleobserver) in the `initializeYoutubePlayer()` method and remove it in the `releaseYoutubePlayer()` method
3. Instead of releasing the player in the `onPause()` method, only store the current playback position
4. Don't remove the view in the `releaseYoutubePlayer()` method